zigzag/classes/stages/RemoveNoUseMemStage.py

```diff
@@ -0,0 +1,176 @@
+        super().__init__(list_of_callables, **kwargs)
+        self.accelerator = accelerator
+        self.layer = layer
+        self.layer_list = layer_list
+        self.mem_update_list = mem_update_list
+        self.mem_update_weight = mem_update_weight
+
+    def run(self) -> Generator:
+        modified_accelerator = self.generate_accelerator_removing_nouse_mem()
+        sub_stage = self.list_of_callables[0](self.list_of_callables[1:],
+                accelerator=modified_accelerator,
+                layer=self.layer,
+                **self.kwargs,)
+        for cme, extra_info in sub_stage.run():
+            yield cme, extra_info
+
+    def generate_accelerator_removing_nouse_mem(self):
+        ## Remove nouse memory level according to update_mem_list and mem_update_weight
+        curr_id = self.layer_list[self.layer]  # current layer id (key) in mem_udpate_list
+        output_operand = self.layer.memory_operand_links[self.layer.output_operand]  # output representation
+        core = next(iter(self.accelerator.cores))
+        operational_array = core.operational_array
+        memory_hierarchy = core.memory_hierarchy
+
+        if len(self.layer.constant_operands) == 1:
+            act_operand = self.layer.memory_operand_links[[operand for operand in self.layer.input_operands if operand not in self.layer.constant_operands][0]]  # act representation
+            const_operand = self.layer.memory_operand_links[self.layer.constant_operands[0]]  # weight representation
+        elif len(self.layer.constant_operands) == 0:
+            # special case when defining workload manually:
+            # the constant operands list is empty for such as "Adder" layers
+            # for input operand, we will represent all inputs as one input, since only their data size is used for required mem size calculation.
+            act_operand = self.layer.memory_operand_links[self.layer.input_operands[0]]
+            const_operand = self.layer.memory_operand_links[self.layer.input_operands[1]]
+        else:
+            # special case when defining workload manually:
+            # both I and W are considered as constant operands for the first layer
+            pr_loop_keys = tuple(self.layer.pr_loop.keys())
+            for operand, related_loop in self.layer.operand_dimensionality_order.items():
+                if pr_loop_keys[0] in related_loop:
+                    act_operand = operand
+            weight_operand: list = [x for x in self.layer.constant_operands if x != act_operand]
+            assert len(weight_operand) == 1
+            weight_operand: str = weight_operand[0]
+            act_operand = self.layer.memory_operand_links[act_operand]  # map from layer representation to hardware representation
+            const_operand = self.layer.memory_operand_links[weight_operand]  # weight representation
+
+        # Find target_act/const/output_mem_level
+        for pos, ele in enumerate(self.mem_update_list[f"{curr_id}"]):
```

Comment:
is curr_id not already a string? If not, you should convert a variable to a string using str(...)